scripts/open-coredump.sh

```diff
@@ -0,0 +1,291 @@
+re-use what is already donwloaded. It is *strongly* recommended to run
+this from an empty directory, with nothing but the core-file present.
+
+Options:
+
+--help,-h
+    Print this help message and exit.
+
+--quiet,-q
+    Only print errors (if any).
+
+--progress
+    Print progress information when downloading the package and cloning
+    the git repo.
+
+--artifact-dir,-d ARTIFACT_DIR
+    Directory where the scrip will store all downloaded artifacts. If
+    the directory doesn't exist, it will be created.
+    It is strongly recommended that this is an empty directory, not used
+    for anything else, to avoid interference with the script.
+    To reset the state of the script, just delete this directory.
+    This can be the same as the directory the coredump is already in, or
+    some other path.
+    By default, a directory called open-coredump.sh.dir will be created in
+    the script's working directory.
+    Can be also provided via env variable ARTIFACT_DIR.
+    If both are provided command line has precedence.
+
+--scylla-s3-reloc-server-url SCYLLA_S3_RELOC_SERVER_URL
+    The URL of the s3 reloc server to connect to. Needed to fetch the
+    build info based on the build-id.
+    Defaults to http://backtrace.scylladb.com.
+    Can be also provided via env variable SCYLLA_S3_RELOC_SERVER_URL.
+    If both are provided command line has precedence.
+
+--scylla-repo-path SCYLLA_REPO_PATH
+    Path to an existing scylla repository to use. The repo is expected to have
+    the branch and commit the packages were built from.
+    If not provided, the appropriate git repository (either scylladb.git or
+    scylla-enterprise.git) will be cloned from https://github.com/scylladb.
+    The script assumes you have the appropriate access rights set up.
+    Can be also provided via env variable SCYLLA_REPO_PATH.
+    If both are provided command line has precedence.
+
+--scylla-gdb-py-source repo|package|none
+    Where to get the scylla-gdb.py script from:
+    * repo - download the latest one from the appropriate main branch;
+    * package - use the one that comes with the relocatable package;
+    * none - don't try to obtain scylla-gdb.py;
+    Defaults to repo.
+    Unlike other artifacts, scylla-gdb.py is placed in the same directory
+    where the coredump is located. If one is already present, it is not
+    overwritten.
+    Can be also provided via env variable SCYLLA_GDB_PY_SOURCE.
+    If both are provided command line has precedence.
+EOF
+}
+
+function log {
+    if [[ $VERBOSE_LEVEL -ge 1 ]]
+    then
+        echo $@
+    fi
+}
+
+for required in eu-unstrip jq curl git; do
+    if ! type $required >& /dev/null; then
+        echo "error: missing required program $required, please install first"
+        exit 1
+    fi
+done
+
+VERBOSE_LEVEL=1
+SCYLLA_S3_RELOC_SERVER_URL="${SCYLLA_S3_RELOC_SERVER_URL:-http://backtrace.scylladb.com}"
+SCYLLA_REPO_PATH="${SCYLLA_REPO_PATH}"
```

Comment:
I don't understand why this is needed, with no default value if `SCYLLA_REPO_PATH` is not in the environment already.

Reply:
Just for consistency. I wanted to list all the input here.